Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/CustomWebTestConfigurationFactory.cs

```diff
@@ -96,6 +98,26 @@ protected override IBrowserConfiguration CreateCustomBrowserConfiguration (WebTe
       return base.CreateCustomBrowserConfiguration (configSettings);
     }
 
+    private string GetVersionedBrowserFolder (string browserName, string versionArchivePath, string latestTestedMajorBrowserVersion)
+    {
+      return Directory.GetFiles (versionArchivePath)
+          .Select (Path.GetFileName)
+          .Where (x => x.StartsWith (browserName))
+          .ToDictionary (fileName => fileName, fileName => CreateVersionFromBrowserZipName (fileName, browserName))
+          .OrderByDescending (kvp => kvp.Value)
+          .First (kvp => kvp.Value.Major == int.Parse (latestTestedMajorBrowserVersion))
+          .Key;
+    }
+
+    private Version CreateVersionFromBrowserZipName (string fileName, string browserName)
+    {
+      var regex = new Regex (@$"^{browserName}_v(?<major>\d*).?(?<minor>\d*).?(?<patch>\d*).?(?<build>\d*)");

```

Comment:
I'm thinking this should be easier/more compact. Can't we just say, there's either just a major version or a four-part-version and nothing in between? We can always pad with zeros on the right.

I don't know if this would be easier to read, but right now, there's a lot of code for very basic version parsing.

Reply:
Also,  this whole version detection should not be part of the upgrade-Edge PR but rather a separate issue. Don't know if it needs to happen first, but it shoud not be the same PR.